subprojects/core/src/main/java/org/gradle/api/internal/classpath/PluginModuleRegistry.java

```diff
@@ -22,4 +22,6 @@
  */
 public interface PluginModuleRegistry {
```

Comment:
Some documentation here would be nice, i.e. what's the difference between pluginModules and extensionPluginModules. Maybe a better name for this split would be `apiModules` and `implementationModules`.